### PR TITLE
ADO.NET Sample App

### PR DESCRIPTION
**Pull request by jeremyp-bq**
Added an ADO.NET sample app, which has been tested on Windows, Linux (x86_64 and aarch64), and Mac.  The sample app compiles under multiple versions of .NET, including .NET Framework on Windows and a couple different versions of .NET Core on all platforms.

⚠️ This PR should only be merged after Vertica 23.3 has been released.  The current Vertica 12.0.4 ADO.NET client driver does not support .NET Core and is not published to NuGet.

### CODE REVIEW

**Comment by williammatherjones** [comment]
I made some changes to the sample app based on the recent demos I have been doing for the .net core driver.   The SampleApp class has been updated to do basic exception handling and to print out a cleaner form of the result set.   The code that prints result sets has been implemented in a separate class.   This is a cleaner implementation and it avoids distracting people viewing the sample app with code that is not relevant to the sample.

File: ADO.NET/SampleApp.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <Architecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</Architecture>
+    <RuntimeIdentifier Condition="$([MSBuild]::IsOSPlatform('Windows'))">win-$(Architecture)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="$([MSBuild]::IsOSPlatform('Linux'))">linux-$(Architecture)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="$([MSBuild]::IsOSPlatform('OSX'))">osx-$(Architecture)</RuntimeIdentifier>
+    <Configuration>Release</Configuration>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Vertica.Data" Version="23.3.0" />
```
Comment (kevinkarch88) [inline ADO.NET/SampleApp.csproj:16]: Did you mean to change the version to rc0?

Comment (jeremyp-bq) [reply ADO.NET/SampleApp.csproj:16]: I figured the example should point to the release version, even if it isn't out yet, and the note in the README mentions what to change in order to reference pre-release versions, such as `23.3.0-rc`.

Although we could have the sample point to `23.3.0-rc` by default and we could update it to `23.3.0` when the official release is published.

Comment (kevinkarch88) [reply ADO.NET/SampleApp.csproj:16]: I think the way it is now is fine.

**Review by kevinkarch88 — APPROVED** [review]

**Review by williammatherjones — APPROVED** [review]

**Pull request merged**